runtime/common/src/claims.rs

```diff
@@ -391,6 +392,26 @@ decl_module! {
 			Self::process_claim(signer, who.clone())?;
 			Preclaims::<T>::remove(&who);
 		}
+
+		#[weight = (
+			T::DbWeight::get().reads_writes(4, 4) + 100_000_000_000,
+			DispatchClass::Normal,
+			Pays::No
+		)]
+		fn move_claim(origin,
+			old: EthereumAddress,
+			new: EthereumAddress,
+			maybe_preclaim: Option<T::AccountId>,
+		) {
+			T::MoveClaimOrigin::try_origin(origin).map(|_| ()).or_else(ensure_root)?;
+
+			Claims::<T>::take(&old).map(|c| Claims::<T>::insert(&new, c));
+			Vesting::<T>::take(&old).map(|c| Vesting::<T>::insert(&new, c));
+			Signing::take(&old).map(|c| Signing::insert(&new, c));
+			maybe_preclaim.map(|preclaim| Preclaims::<T>::mutate(&preclaim, |maybe_o|
```

Comment:
I might be nice to do this check first, and if `Some(preclaim)` comes in from the input, but we dont find it, we return error, as it might imply some bigger problem is occurring with the input.